backend/controller/cronjobs/cronjobs.go

```diff
@@ -0,0 +1,483 @@
+package cronjobs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	"github.com/TBD54566975/ftl/backend/controller/scheduledtask"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	schemapb "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/schema"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/cron"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/TBD54566975/ftl/internal/slices"
+	"github.com/alecthomas/atomic"
+	"github.com/alecthomas/types/optional"
+	"github.com/alecthomas/types/pubsub"
+	"github.com/benbjohnson/clock"
+	"github.com/jpillora/backoff"
+	"github.com/serialx/hashring"
+	sl "golang.org/x/exp/slices"
+)
+
+const (
+	controllersPerJob              = 2
+	jobResetInterval               = time.Minute
+	newJobHashRingOverrideInterval = time.Minute + time.Second*20
+)
+
+type Config struct {
+	Timeout time.Duration
+}
+
+type jobChangeType int
```

Comment:
Make this a "sum type". It's slightly more work up front, but it's less error-prone. For example the `jobChange` type associates a particular "type" with a field being present, but there's no way to guarantee that is used correctly.